gadget/kcmdline.go

```diff
@@ -23,15 +23,18 @@ import (
 	"fmt"
 
 	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/strutil"
 )
 
 type kargKey struct{ par, val string }
 type kernelArgsSet map[kargKey]bool
 
-// CheckCmdlineAllowed returns an error if an argument from cmdline is
-// not on a list of allowed kernel arguments. A wild card ('*') can be
-// used in the allow list for the values.
-func CheckCmdlineAllowed(cmdline string, allowedSl []osutil.KernelArgument) error {
+// FilterKernelCmdline returns a filtered command line, removing
+// arguments that are not on a list of allowed kernel arguments. A
+// wild card ('*') can be used in the allow list for the
+// values. Additionally, a string with the arguments that have been
+// filtered out is also returned.
+func FilterKernelCmdline(cmdline string, allowedSl []osutil.KernelArgument) (argsAllowed, argsDenied string) {
```

Comment:
It might be cleaner maybe for pathological cases, and then by printing with some care when creating the error returned by `validateCmdlineParamsAreAllowed()`. On the other hand the user sets the option as a single line so they should be able to spot the problem, and the parsing already should clean up some ugliness. But I can do that as a follow up.